controllers/argocd/deployment.go

```diff
@@ -962,6 +962,11 @@ func (r *ReconcileArgoCD) reconcileRepoDeployment(cr *argoprojv1a1.ArgoCD) error
 		desiredImage := getRepoServerContainerImage(cr)
 		if actualImage != desiredImage {
 			existing.Spec.Template.Spec.Containers[0].Image = desiredImage
+			if existing.Spec.Template.ObjectMeta.Labels == nil {
```

Comment:
Is this change also made due to a failing test?

Reply:
Yes, It is made to fix the failing test. I will try to investigate more on what was the reason for unit test failure tomorrow and get back.

Reply:
I can only suspect it has something to do with the kubernetes client version update to `1.22.2`.

Reply:
However, I figured out the root cause of the issue.
The latest version of controller-runtime overrides the object instead of updating.

Please refer to the below PR.
kubernetes-sigs/controller-runtime#1651